openlibrary/plugins/worksearch/subjects.py

```diff
@@ -47,27 +47,22 @@ class subjects(delegate.page):
     path = '(/subjects/[^/]+)'
 
     def GET(self, key):
+        from openlibrary.plugins.worksearch.code import search, subject_search
+
+        # Make sure the /subjects url looks right
         nkey = self.normalize_key(key)
         if nkey != key:
             raise web.redirect(nkey)
 
-        # this needs to be updated to include:
-        #q=public_scan_b:true+OR+lending_edition_s:*
-        subj = get_subject(key, details=True, filters={
-            'public_scan_b': 'false',
-            'lending_edition_s': '*'
-        })
+        # Nauhty hack to make 'subject' work like 'person', etc
+        if ':' not in key:
+            key = key.replace('/subjects/', '/subjects/subject:', 1)
 
-        subj.v2 = True
-        delegate.context.setdefault('bodyid', 'subject')
-        if not subj or subj.work_count == 0:
+        subjs = subject_search.get_results('key:"%s"' % key).get('response', {}).get('docs')
```

Comment:
If there's no `response` there will be no `docs` either and this will error. I'm seeing more and more of these types of intermittent errors on the web site when the search engine fails to produce results.